packages/asset/contracts/AssetMinter.sol

```diff
@@ -110,20 +112,21 @@ contract AssetMinter is
             mintableAsset.amount,
             mintableAsset.tier,
             mintableAsset.creatorNonce,
-            mintAsRevealed,
-            0
+            mintAsRevealed
```

Comment:
Seems like there is a problem, here we pass the `mintAsRevealed` but in the `Asset` contract mint function we're not using that correctly and even for the common tier we are minting as unrevealed.

In Asset, in the mint function we need to change id generation from `0`, to the value from `mintAsRevealed`

```
        uint256 id = TokenIdUtils.generateTokenId(
            assetData.creator,
            assetData.tier,
            nonce,
            0
        );
        ```